src/drivers/char/terminal/vtparse.c

```diff
@@ -46,10 +46,11 @@ static void do_action(struct vtparse *parser, vt_action_t action, char ch) {
 		break;
 
 	case VT_ACTION_COLLECT:
-		if (token->attrs_len == -1) {
+		// Buffer overflow check may be needed
+		//if (token->attrs_len == -1) {
```

Comment:
As far as I understood, attr overflow checked in other states (attrs buffer is VTPARSE_TOKEN_PARAMS_MAX (2) long). If it overflows, then len is assigned to `-1` and this as guard conditions to all states.
The problem seems to be that `attrs_len` is of `char` type, which 
a) could be signed or unsigned
b) is not what you'd expect for length type.

I'd suggest to revert this code and change attrs_len to `int` (probably reorganized vt_token structure to  get rid of unnecesary member padding)

@abusalimov could have something to say (as the author)